File: src/main/java/com/key/oa/entity/Employee.java
package com.key.oa.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.google.common.base.Objects;
import com.key.oa.annotation.Desensitize;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

/**
 * @author 孙强
 * 员工实体类
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    @Desensitize(strategy = Desensitize.DesensitizeStrategy.PHONE)
    private String phone;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false, unique = true)
    @Desensitize(strategy = Desensitize.DesensitizeStrategy.ID_CARD)
    private String identity;

    @Temporal(TemporalType.DATE)
    @Column(nullable = false)
    private Date birthday;

    /**
     * 员工的工号
     */
    @Column(nullable = false, unique = true)
    private String jobNumber;

    @Column(nullable = false)
    @Desensitize(strategy = Desensitize.DesensitizeStrategy.PASSWORD)
    private String password;

    @Column(nullable = false)
    private Boolean verified;

    @JsonIgnore
    @ManyToMany(mappedBy = "employees")
    private Set<Role> roles = new HashSet<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Employee employee)) {
            return false;
        }
        return Objects.equal(getId(), employee.getId()) && Objects.equal(getIdentity(), employee.getIdentity()) && Objects.equal(getJobNumber(), employee.getJobNumber());
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(getId(), getIdentity(), getJobNumber());
    }
}

File: src/main/java/com/key/oa/entity/Employee.kt
package com.key.oa.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import com.key.oa.annotation.Desensitize
import jakarta.persistence.*
import java.util.*
import kotlin.collections.HashSet

@Entity
class Employee(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long,

    @Column(nullable = false) var name: String,

    @Desensitize(strategy = Desensitize.DesensitizeStrategy.PHONE)
    @Column(nullable = false, unique = true) var phone: String,

    @Column(nullable = false, unique = true) var email: String,

    @Column(nullable = false, unique = true) var identity: String,

    @Column(nullable = false) var birthday: Date,

    @Column(nullable = false, unique = true) var jobNumber: String,  // 员工的工号

    @Desensitize(strategy = Desensitize.DesensitizeStrategy.PASSWORD)
    @Column(nullable = false) var password: String,

    @Column(nullable = false) var verified: Boolean,

    @JsonIgnore
    @ManyToMany(mappedBy = "employees")
    var roles: MutableSet<Role> = HashSet()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Employee) return false

        if (id != other.id) return false
        if (identity != other.identity) return false
        if (jobNumber != other.jobNumber) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + identity.hashCode()
        result = 31 * result + jobNumber.hashCode()
        return result
    }
}